packages/runtime/src/controls/rich-text/introspection.ts

```diff
@@ -84,7 +84,9 @@ export function getRichTextPageIds(value: RichTextControlData) {
     switch (inline.type) {
       case 'link': {
         const nodePageIds = inline.nodes?.flatMap(getNodePageIds) ?? []
-        const dataPageIds = inline.data ? getLinkDataPageIds(inline.data as LinkValue) : []
+        const dataPageIds = inline.data
+          ? getLinkPropControllerPageIds(inline.data as LinkPropControllerData)
```

Comment:
We should be casting this to `Link` instead, no? Since there won't be multiple versions and data types for the prop controller here?